src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -337,17 +345,29 @@ Type[] getAllGenericParameterTypes() {
             if (realParamData) {
                 final Type[] out = new Type[nonGenericParamTypes.length];
                 final Parameter[] params = getParameters();
-                int fromidx = 0;
-                for (int i = 0; i < out.length; i++) {
-                    final Parameter param = params[i];
-                    if (param.isSynthetic() || param.isImplicit()) {
-                        // If we hit a synthetic or mandated parameter,
-                        // use the non generic parameter info.
-                        out[i] = nonGenericParamTypes[i];
+                if (getDeclaringClass().isRecord() && this instanceof Constructor) {
```

Comment:
Sorry for late notice but we can move the 2 variable allocation to the else block, to speed up the record constructor path.

Reply:
yep true, will do